### ISSUE TITLE
Bracketed arguments adding space in ClickHouse dialect

### ISSUE THREAD

**Issue by Raquel10-RevenueCat** [sqlfluff/sqlfluff#6532]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I get this error when running sqlfluff with ClickHouse dialect:

```
L: 238 | P:  65 | LT01 | Expected single whitespace between data type identifier
                       | and start bracket '('. [layout.spacing]
```

This is an example line that is being linted incorrectly. Original line:
```
    event_created_at DateTime64(3),  -- ms precision
```

Linted line:
```
    event_created_at DateTime64 (3),  -- ms precision
```

I'd expect that the line is not modified. I am using the default configuration

### Expected Behaviour

I'd expect that the line is not modified

### Observed Behaviour

This is an example line that is being linted incorrectly. Original line:
```
    event_created_at DateTime64(3),  -- ms precision
```

Linted line:
```
    event_created_at DateTime64 (3),  -- ms precision
```

### How to reproduce

Add any snippet with a datatype and bracketed args, lint with clickhouse dialect and a space is introduced 

### Dialect

clickhouse

### Version

3.3.0

### Configuration

[sqlfluff]

dialect = clickhouse

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by enqueue** [comment]
It does not happen with every data type, e.g. `Decimal(18,9)` is fine.

Here is another example:

```sql
create table `ecb_exchangerates` (
    report_time DateTime default now(),
    currency FixedString(3),
    currency_denominator LowCardinality(FixedString(3)),
    time_period Date,
    obs_value Decimal(18, 9),
    obs_status Enum8('H' = 0, 'A' = 1)
)
engine = MergeTree
order by (toStartOfMonth(report_time));
```

Workaround: Adjust individual configuation:

```
[sqlfluff:layout:type:data_type]
spacing_within = touch:inline
```
